### ISSUE TITLE
Learning rate scheduler is called twice on both train and test losses

### ISSUE THREAD

**Issue by RMeli** [RMeli/gnina-torch#23]
PR #10 asserts that "the learning rate update is based on the loss function of the test set (instead of the training set)". This was the intent (although it should probably monitor the loss function of the training set for consistency), but by attaching the LR scheduler update to the evaluator

https://github.com/RMeli/gnina-torch/blob/f8aecb6688ee359ee9499f0cb3019094ec7058a0/gnina/training.py#L652-L656

effectively causes the LR scheduler to be called twice, since the same evaluator is used to evaluate the training set and the test set.

https://github.com/RMeli/gnina-torch/blob/f8aecb6688ee359ee9499f0cb3019094ec7058a0/gnina/training.py#L630-L632

https://github.com/RMeli/gnina-torch/blob/f8aecb6688ee359ee9499f0cb3019094ec7058a0/gnina/training.py#L677-L679

Possible solutions suggested by @vfdev-5:
* Use if/else to check which data loader has been used by the evaluator
* Use custom filtering (decorated if/else) to check which data loader has been used by the evaluator
* Register custom events (to distinguish between train and test evaluation)
* Split into two distinct evaluators
* Call the scheduler update directly in the interested evaluation function
